src/backend/binaryen/wasm_expr_gen.ts

```diff
@@ -2033,150 +2120,142 @@ export class WASMExpressionGen {
-            res = this.getClosureOfMethod(func, methodType);
-        }
-        return res;
-    }
 
     private getInfcMember(
         member: MemberDescription,
         infcType: ValueType,
         thisRef: binaryen.ExpressionRef,
-        memberIdx: number,
+        valueIdx: number,
+        metaRef: binaryen.ExpressionRef,
+        indexRef: binaryen.ExpressionRef,
+        flagRef: binaryen.ExpressionRef,
+        propTypeIdRef: binaryen.ExpressionRef,
         isCall = false,
         args?: SemanticsValue[],
+        isSetter = false,
     ) {
-        let res: binaryen.ExpressionRef;
-        const memberValueType = member.hasGetter
+        const propType = isSetter
+            ? (member.setter as VarValue).type
+            : member.hasGetter
             ? (member.getter as VarValue).type
             : member.valueType;
 
-        res = this.getOriObjInfoByFindIdx(
+        /* TODO: workaround: quick path may fail, since cast failure */
+        const infcDescTypeRef = this.wasmTypeGen.getWASMObjOriType(infcType);
+        const castedObjRef = binaryenCAPI._BinaryenRefCast(
+            this.module.ptr,
             thisRef,
-            infcType,
-            member,
-            memberValueType,
-            memberIdx,
-            false,
-            member.type !== MemberType.FIELD,
+            infcDescTypeRef,
+        );
+        const ifEqualTypeId = FunctionalFuncs.isTypeIdEqual(
+            this.module,
+            infcType.typeId,
+            metaRef,
         );
-        /** is call method */
+        let ifEqualTrue: binaryen.ExpressionRef;
+        if (propType.kind === ValueTypeKind.FUNCTION) {
+            /* if property's value type is function, and typeid is equal, then we can get property from vtable */
+            /* methodRef get from vtable is a funcref, we need to box it to closure */
+            ifEqualTrue = this.getClosureOfMethod(
+                this.getObjMethod(castedObjRef, valueIdx, infcDescTypeRef),
+                propType as FunctionType,
+            );
+        } else {
+            /* if property's value type is not function, then it must be a field */
+            ifEqualTrue = this.getObjField(
+                castedObjRef,
+                valueIdx,
+                infcDescTypeRef,
+            );
+            // TODO: box & unbox depend on field_type_id
+        }
+        const ifEqualFalse = this.dynGetInfcProperty(
+            thisRef,
+            indexRef,
+            flagRef,
+            propType,
+            member.isOptional,
+            propTypeIdRef,
+        );
+        /* get property from interface */
+        let res = this.module.if(ifEqualTypeId, ifEqualTrue, ifEqualFalse);
+
+        /* If isCall or member is an accessor, call the memberRef, and get the result */
         if (
             member.type === MemberType.ACCESSOR ||
-            (member.type === MemberType.METHOD && isCall)
+            (propType.kind === ValueTypeKind.FUNCTION && isCall)
```

Comment:
Why is it a FUNCTION rather than METHOD?

Reply:
`FUNCTION` is its value type _(ValueTypeKind.FUNCTION)_, and `METHOD` is its member type _(MemberType.METHOD, to flag it is a method, not a field, which stores in vtable, not in instance)_.

Use the value type as the judgement, we can contain all situations, like 
```ts
const obj: I= {
        a: 5,
        x: () => 1,
    }
obj.x();
```